master/buildbot/test/unit/test_db_migrate_versions_024_add_buildslaves_table.py

```diff
@@ -42,9 +42,15 @@ def verify_thd(conn):
             self.assertEqual(res.fetchall(), [])
 
             # and buildslave name is unique, so we'll get an error here
-            dialect = conn.dialect.name
-            exc = (sa.exc.ProgrammingError if dialect == 'postgresql'
-                   else sa.exc.IntegrityError)
+            if conn.dialect.name == 'postgresql':
```

Comment:
It depends on answer to the following question: if unique constraint is being broken, is it more like ProgrammingError, or IntegrityError?

If it is semantically IntegrityError, then by default IntegrityError should be used, and special case pg8000 should be added (as you suggested).